### ISSUE TITLE
Poor performance

### ISSUE THREAD

**Issue by tokudaek** [KranthiGV/Pretrained-Show-and-Tell-model#6]
First, thanks for sharing your work. Results using these models though seems to be worse than Karpathy's work.

**Comment by KranthiGV** [comment]
@tokudaek 
I will evaluate the results and get back to you.

**Comment by KranthiGV** [comment]
@tokudaek 
The graphs seems to be okay. 
I accept that this model doesn't perform well on all types of images.
The issue can also be with the 2M checkpoint (I can run it at least until 3M) in which Inception weights are also updated. So, the loss is still high at 2M iteration. (I observed a peak in loss after 1M iteration when I started updating weights of CNN and LSTM together.)
Can you try to use 1M checkpoint file? You **might** get better results.

Let me know if I'm missing anything. :smile:

**Comment by ericktokuda** [comment]
Nice, I ll try testing the 1M checkpoint. Thanks a lot.

**Comment by ankush-06** [comment]
To be really honest, i could'n really solved that issue :) but i somewhat overcame that. The problem was that the program was not able to locate the "word_counts.txt" file. I tried everything. I was getting frustated. So, i ran the commands in a separate python file("xxx.py"): 

_with tf.gfile.GFile("word_counts.txt") as f:
      word_map = list(f.readlines())
word_map = [line.split()[0] for line in word_map]_

then i imported this word_map list in the vocabulary.py file.
_from xxx import word_map
reverse_vocab = word_map_

**Comment by peri044** [comment]
Hey @KranthiGV  I'm trying to train the im2txt model and no custom code written. 
![capture](https://user-images.githubusercontent.com/10511428/28548832-3e31b9f6-70a4-11e7-87a5-cc04441751bf.PNG)

This is the loss curve during training. The model so far trained only for 310,000 iterations. The loss has not been changing much over the last 200,000 iterations. It is around 2-2.3. The captions generated for validation images seem good. But I suspect the loss is high. Is this the same behavior you see in your training?

**Comment by stephanedebove** [comment]
For me the 2M model works better than the 1M, but it's still far from perfect (takes an accordeon for a tie for instance). Is there any way to improve the model? Would going to 3M help?
